### PR TITLE
fix: fail closed when canonical snapshot unavailable

### LINKED ISSUE #293: [Bug] Do not advertise canonical-write when Export XML snapshot is unavailable
## Summary

The live Final Cut canonical provider advertises canonical-write support even when its canonical snapshot path cannot open the Export XML window.

## Expected behavior

Capability discovery must not advertise canonical-read or canonical-write until a supported canonical snapshot read is available. If the snapshot provider is unavailable, MCP should return a structured unavailable capability and route canonical edits away from the editor path.

## Actual behavior

editor.inspect reported canonicalTimelineMode as canonical-write, canonicalDocument.read and canonicalDocument.write as available, and editing.route selected the editor path for timeline.edit. The subsequent project.inspect and timeline.inspect calls failed with FINAL_CUT_CANONICAL_EXPORT_WINDOW_UNAVAILABLE.

## Reproduction

1. Start the merged main MCP server in headed native mode with FRAMEKIT_EDITOR=final-cut-live, FRAMEKIT_FINAL_CUT_CANONICAL_PROVIDER=native, and FRAMEKIT_FINAL_CUT_NATIVE_WRITES=1.
2. Call editor.inspect and editing.route with operation timeline.edit.
3. Call project.inspect or timeline.inspect.
4. Observe that capability discovery claims canonical-write while the canonical snapshot read fails.

## Environment

- Framekit commit/version: f1cee2516b41fa8e7c22148954bfd3097d99aa1f / 0.1.8
- Codex/MCP setup: fresh stdio MCP process from merged main
- Editor and version: Final Cut Pro; version Needs confirmation
- Operating system: macOS; version Needs confirmation
- Runtime mode: headed

## Evidence

- editor.inspect reported the final-cut-native-canonical backend with canonical-write.
- editing.route reported editor-selected with no missing capabilities.
- project.inspect and timeline.inspect failed with FINAL_CUT_CANONICAL_EXPORT_WINDOW_UNAVAILABLE.
- No timeline mutation was executed.

## Safety impact

This creates a dangerous false-positive capability boundary: a client can be told that canonical editing is available even though the required canonical read and revision guard cannot be obtained. The implementation must fail closed before mutation.

## Additional context

Related to #194. The current-version Final Cut session was the disposable Framekit Native E2E project.

### PR DESCRIPTION
**Pull request by morshoto**
- [x] Request PR review
- [ ] If you are unable to review, please set it as a [PR draft](https://github.blog/2019-02-14-introducing-draft-pull-requests/) draft.
- [x] Github issue: Closes: #293

## Summary

Canonical Final Cut capability discovery now probes the Export XML snapshot path before advertising canonical read/write support. Export-window failures downgrade the provider to `metadata-only`, preserve the structured unavailable reason through session composition, and prevent canonical editing routes or native mutation from proceeding.

## Validation

```bash
pnpm install --frozen-lockfile
pnpm run build
pnpm run test
pnpm run check:boundaries
```

All deterministic gates passed: 905 tests passed, 1 intentional native test skipped. Native Xcode checks were not required because no native files changed.

## Architecture and compatibility

- [x] Runtime remains independent of MCP and editor-specific implementations.
- [x] Public capability inspection options remain backward compatible; background project listing can skip canonical UI probes.
- [x] Existing adapters and test fixtures remain compatible.

## Safety

- [x] No credentials, private media, raw crash dumps, or user-specific local paths were added.
- [x] Generated files and build artifacts are excluded.
- [x] No documentation or environment-variable changes were required.

### CODE REVIEW

**Comment by morshoto** [comment]
Review complete against issue #293: no [must] or [nits] findings. The canonical snapshot probe, metadata-only downgrade, session capability propagation, route fail-closed behavior, and no-mutation MCP regression coverage align with the issue, and all required checks pass. GitHub does not allow the PR author to submit an approval review.

File: adapters/final-cut/typescript/src/session.ts [hunk 1]
```diff
@@ -105,7 +106,7 @@ export class FinalCutSessionAdapter implements EditorPort, LiveEditorStatePort {
       : liveMutation
         ? this.options.live
         : this.options.snapshot;
-    const readCapabilities = snapshot ?? (liveSnapshot ? live : undefined);
+    const readCapabilities = snapshot ?? (useLiveCanonical ? live : undefined);
```
Comment (morshoto) [inline adapters/final-cut/typescript/src/session.ts]: [must] Keep `readCapabilities` gated by `liveSnapshot` (or otherwise require `live.readProject`) here. `FinalCutSessionOptions.live` intentionally makes `readProject` optional; with a live adapter that reports `timelineSnapshotRead` but does not implement `readProject`, this assignment still advertises the live read families and `canonicalDocument.read` even though `readAdapter` is undefined and `FinalCutSessionAdapter.readProject()` will throw `CAPABILITY_UNAVAILABLE`. That recreates the capability/`project.inspect` mismatch this PR is meant to remove. Please add a regression test for a live adapter without `readProject` and keep canonical read unavailable.

**Comment by morshoto** [comment]
Review update: an additional [must] finding is posted inline on `FinalCutSessionAdapter.getCapabilities`. A live adapter without `readProject` can still be advertised as canonical-readable because `readCapabilities` is not gated by `liveSnapshot`, while `project.inspect` then fails. Please add that regression case and keep the capability unavailable. GitHub does not allow the PR author to submit a request-changes review.

File: adapters/final-cut/typescript/src/session.ts (hunk 1 above)
Comment (morshoto) [reply adapters/final-cut/typescript/src/session.ts]: Fixed in a60eb8e. Added a regression covering a live adapter that reports canonical snapshot capability without readProject; the session now gates read capabilities on liveSnapshot, so canonical reads remain unavailable. Affected tests pass 53/53, and the commit hook passed build, full tests, and boundaries.

**Comment by morshoto** [comment]
Re-review of head a60eb8e: the [must] finding is fixed. `readCapabilities` is gated by `liveSnapshot`, and the new regression covers a live adapter without `readProject`; the affected tests/build/boundaries passed per the author. Remaining remote checks are still pending, so final readiness is subject to those checks.
